### PR TITLE
Revamping API tests on canned data

### PR DESCRIPTION
**Pull request by giorgiosironi**
- Tests for 404 on item are working
- Checking the data generation script is successful
- Added all content types that already pass tests. The randomized data is validating with the current schema for these.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Locally the uncommented tests passed, but on ci they don't; probably a difference in content, resetting it locally.

**Comment by giorgiosironi** [comment]
Things like `items[0].selectedCurator.orcid` need to be customized to generate a correct ORCID.

File: src/modules/jcms_rest/tests/src/Functional/RamlSchemaValidationTest.php [hunk 1]
```diff
@@ -12,30 +12,47 @@
...
+      self::$contentGenerated = TRUE;
+      $projectRoot = realpath(__DIR__ . '/../../../../../..');
+      $script = $projectRoot . '/scripts/generate_content.sh';
+      if (!file_exists($script)) {
+        throw new RuntimeException("File $script does not exist");
+      }
+      $logFile = '/tmp/generate_content.log';
+      exec("$script >$logFile 2>&1", $output, $exitCode);
+      if ($exitCode != 0) {
+        throw new RuntimeException("$script failed. Check log file $logFile");
```
Comment (thewilkybarkid) [inline src/modules/jcms_rest/tests/src/Functional/RamlSchemaValidationTest.php]: `symfony/process` is available, can simplify this.

File: sync/field.field.node.annual_report.field_annual_report_year.yml [hunk 2]
```diff
@@ -17,7 +17,9 @@ default_value: {  }
 default_value_callback: ''
 settings:
   min: 2012
-  max: null
+  # default maximum value is 999 in:
+  # ./web/core/lib/Drupal/Core/Field/Plugin/Field/FieldType/IntegerItem.php
+  max: 2099
```
Comment (nlisgo) [inline sync/field.field.node.annual_report.field_annual_report_year.yml]: Is this prophetic?

File: src/modules/jcms_rest/tests/src/Functional/RamlSchemaValidationTest.php (hunk 1 above)
Comment (giorgiosironi) [reply src/modules/jcms_rest/tests/src/Functional/RamlSchemaValidationTest.php]: It is not in the explicit requirements in composer.json, so I would avoid introducing additional dependencies in place of 3-4 lines of standard library calls

File: sync/field.field.node.annual_report.field_annual_report_year.yml (hunk 2 above)
Comment (giorgiosironi) [reply sync/field.field.node.annual_report.field_annual_report_year.yml]: Space: 2099

**Pull request merged**